### PR TITLE
Compress the trace log #1475

### PR DESCRIPTION
**Pull request by soulgalore**

### CODE REVIEW

File: lib/plugins/tracestorer/index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = {
       case 'browsertime.chrometrace':
       case 'webpagetest.chrometrace':
         {
-          this.storageManager.writeDataForUrl(JSON.stringify(message.data, null, 0), message.name, message.url);
```
Comment (tobli) [inline lib/plugins/tracestorer/index.js]: Good catch! I'd expect Bluebird to warn about this. Maybe we didn't run it enough in non-prod mode to notice.

File: lib/support/storageManager.js [hunk 2]
```diff
@@ -6,16 +6,29 @@ const fs = require('fs-extra'),
...
   return Promise.join(dirPath, filename, data,
-    (dirPath, filename, data) =>
-      fs.writeFileAsync(path.join(dirPath, filename), data, 'utf8'));
+    (dirPath, filename, data) => {
+      if (gzip) {
+        const buff = new Buffer(data, 'utf8');
+        return zlib.gzipAsync(buff, {level: 1}).then((buffer) =>
+          fs.writeFileAsync(path.join(dirPath, filename + '.gz'), buffer, 'utf8')
+        );
+      }
```
Comment (tobli) [inline lib/support/storageManager.js]: Just a nitpick, but this isn't how we typically place the braces for else, is it?

Comment (soulgalore) [inline lib/support/storageManager.js]: I'll fix that!

File: lib/plugins/tracestorer/index.js (hunk 1 above)
Comment (soulgalore) [inline lib/plugins/tracestorer/index.js]: it warn in some cases where it is obvious I think.

Comment (tobli) [inline lib/plugins/tracestorer/index.js]: It's supposed to warn if a Bluebird promise was created, but not returned. Anyways, good that this is fixed.

http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-created-in-a-handler-but-was-not-returned-from-it

**Pull request merged**